### DISCUSSION TITLE
How do I "update" downloaded games?

### DISCUSSION THREAD

**Discussion by LucaNerlich** [Kalanyr/gogrepo#116] [Q&A] [answered]
Hi Community,

assuming the following workflow:

1. `gogrepoc login`
2. `gogrepoc update -full`
3. `gogrepoc download`

Ive now successfully downloaded all 2TB of my gog library as an offline backup.

Question: How do I now "update" the downloaded games - as in, Games regularly recieve new versions, which Id like to download as well.
So, how do I now download this delta? Ive just reran `gogrepoc update -full && gogrepoc download` assuming it would 'see' the manifest and compare it to the downloaded files, downloading only the updated binaries now. However, gogrepoc now proceeds to fully re-download my whole library of 2tb.

If i am not complelty blind, `gogrepoc download` now also deleted my old, already downloaded installers?

What am I doing wrong or misunderstanding? Thanks!

**Comment by Kalanyr** [comment]
Assuming your system uses the OS/Lang you want:

-full should only be used rarely ( no more than monthly).  The default
behaviour is to update only new & updated games, full forces it to check
every game in your library.

Are you specifying a folder when you run Download ? Download works by
building a list of everything that needs downloading then skipping ( "pass"
) that already exists.

**Comment by LucaNerlich** [comment]
`python3 ./gogrepoc.py download . -lang en de -os windows mac linux`

This is the command im running.

And this did trigger a full redownload of my whole library - and additionally the previously downloaded installer seem to be gone.

All game folders now only contain an images folder and an info txt. file.
![image](https://github.com/user-attachments/assets/8a7e706a-d973-4cae-b5b4-144f6ea0f8c0)

`!downloading` only contains the installer which have so far now been re-downloaded.

**Comment by Kalanyr** [comment]
Check for a folder called !orphaned , your old installers may be in there.

I strongly recommend against using . as the download directory, especially
if it's a drive / partition / share root , use a folder called GOGDownloads
or something.

**Comment by LucaNerlich** [comment]
`!orphaned` is empty.

im executing the script directly on my backup external harddrive in a folder called `gog-games`. `.` as a download target should be fine. Thanks though 👍

**Comment by Kalanyr** [comment, marked as answer]
I don't know what's going on then, GOGrepoc should never delete files
unless you specify flags to do so ( orphaning is the default behaviour),
there is an exception where if a file with the same name exists it can be
moved to ! downloading as the base to do a partial download but thats a
fairly rare circumstance.

If you have the log files you could try uploading those and I'll see if
they tell me anything useful ?

**Reply by LucaNerlich** [reply]
I do not have any logs, sorry. For now, ill just mark your response as the answer. If I ever find out more, ill re-reply here. Thanks!
